### PR TITLE
feat: [CCM-7784]: Add Search in Cost Categories Group By

### PR DESCRIPTION
**Pull request by rajarshee-harness**
### Summary

This PR adds search functionality in cost categories group by and sorts the list items alphabetically.

#### Screenshots

| Before | After |
|:---:|:---:|
| Screenshot 2022-08-04 at 10 14 22 AM | Screenshot 2022-08-04 at 10 14 13 AM |
| Screenshot 2022-08-04 at 11 46 35 AM | Screenshot 2022-08-04 at 11 46 26 AM |

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 11 Issues 
* Code issue 11 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10385)

**Comment by rajarshee-harness** [comment]
retrigger coverage

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/__test__/GroupByView.test.tsx [hunk 1]
```diff
@@ -53,6 +53,12 @@ const ClusterData = {
   __typename: 'QLCEViewFieldIdentifierData'
 }
 
+jest.mock('framework/strings', () => ({
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveExplorerGroupBy/__test__/GroupByView.test.tsx:56]: Why we mocked this? It already gives the key only

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss [hunk 2]
```diff
@@ -88,14 +73,53 @@
   }
 }
 
+.noResults {
+  margin: 12px;
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss:77]: Could use standard spacing here

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss [hunk 3]
```diff
@@ -88,14 +73,53 @@
...
+.search {
+  margin: var(--spacing-small);
+
+  :global {
+    .bp3-input {
+      color: var(--grey-900);
+      box-shadow: none !important;
+      border: none !important;
+      background-color: var(--grey-100) !important;
+    }
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss:110]: Why was this needed? Not using standard design?

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss [hunk 4]
```diff
@@ -88,14 +73,53 @@
...
+  }
+}
+
+.listItem {
+  color: var(--grey-1000);
+  padding-left: var(--spacing-4);
+  padding-right: var(--spacing-4);
+  font-size: 13px;
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss:118]: Use standard font size if possible

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.tsx [hunk 5]
```diff
@@ -117,21 +138,20 @@ interface LabelDropDownProps {
...
   const [searchText, setSearchText] = useState('')
 
-  const filteredResults = values.filter(val => val.includes(searchText))
+  const filteredResults = values.filter(val => val.toLowerCase().includes(searchText.toLowerCase()))
   return (
-    <section className={css.labelDropDownContainer}>
-      <TextInput
-        className={css.search}
+    <section>
+      <ExpandingSearchInput
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.tsx:148]: Any benefit of using `ExpandingSearchInput` over `TextInput`.

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.tsx [hunk 6]
```diff
@@ -117,21 +138,20 @@ interface LabelDropDownProps {
...
         autoFocus
-        placeholder="Search Value"
-        onChange={e => {
-          const event = e as any
-          const value = event.target.value
-          setSearchText(value)
-        }}
+        alwaysExpanded
+        placeholder={getString('ce.perspectives.createPerspective.filters.searchValue')}
+        onChange={text => setSearchText(text)}
```
Comment (jenil-harness) [inline src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.tsx:153]: Unnecessary function, use `setSearchText` directly

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss (hunk 3 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.module.scss:110]: Was trying to match the styles to the `DropDown` component we have in UICore, I'll revert them.

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.tsx (hunk 5 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/components/PerspectiveExplorerGroupBy/GroupByView.tsx:148]: Was told to add a search icon with the placeholder text, `ExpandingSearchInput` already had the icon.

File: src/modules/75-ce/components/PerspectiveExplorerGroupBy/__test__/GroupByView.test.tsx (hunk 1 above)
Comment (rajarshee-harness) [reply src/modules/75-ce/components/PerspectiveExplorerGroupBy/__test__/GroupByView.test.tsx:56]: Removed this and added `TestWrapper` to the component.

**Review by jenil-harness — APPROVED** [review]

**Review by rishik30 — APPROVED** [review]
